app/models/megaphone/cuepoint.rb

```diff
@@ -8,33 +8,36 @@ class Cuepoint
 
     CREATE_REQUIRED = %i[cuepoint_type ad_count start_time ad_sources]
 
-    CREATE_ATTRIBUTES = CREATE_REQUIRED + %i[title end_time action is_active offset notes]
+    # max_duration is undocumented, but it works to set as a # of seconds
+    CREATE_ATTRIBUTES = CREATE_REQUIRED + %i[title end_time action is_active offset notes max_duration]
 
     ALL_ATTRIBUTES = CREATE_ATTRIBUTES
 
     attr_accessor(*ALL_ATTRIBUTES)
 
     validates_presence_of CREATE_REQUIRED
 
-    def self.from_zones_and_media(zones, media)
+    def self.from_placement_and_media(placement, media)
       cuepoints = []
       current_cuepoint = nil
       original_duration = 0
       original_count = 0
-      zones.each do |zone|
+      placement.zones.each do |zone|
         # if this is an ad zone, add it to the cue point
-        if ["ad", "sonic_id", "house"].include?(zone[:type])
+        if ["ad", "house", "sonic_id", "billboard"].include?(zone[:type])
```

Comment:
Might be slightly more future-proof to just check for `"original"`, since you're including everything else.

Reply:
yah, I had the same thought, then I got lazy - will fix